modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/tabletoolbar/plugin.js

```diff
@@ -0,0 +1,1475 @@
+				},
+				createDef({
+					requiredContent: "table",
+					exec(editor) {
+						const selection = editor.getSelection();
+						insertCell(selection, true);
+					},
+				})
+			);
+
+			addCmd(
+				{
+					commandName: "cellInsertAfter",
+				},
+				createDef({
+					requiredContent: "table",
+					exec(editor) {
+						const selection = editor.getSelection();
+						insertCell(selection);
+					},
+				})
+			);
+		},
+		getSelectedCells,
+	});
+
+	ListBox = CKEDITOR.tools.createClass({
+		$: function (editor, config, clickFn) {
+			this.$ = new CKEDITOR.dom.element("div");
+
+			this.clickFn = clickFn;
+
+			this.setHtml(config);
+		},
+		proto: {
+			getElement: function () {
+				return this.$;
+			},
+
+			getHtml: function () {
+				return this.getElement().getOuterHtml();
+			},
+
+			setHtml: function (config) {
+				this.getElement().setHtml(
+					"<ul class='listbox' id='" +
+						config.panelId +
+						"' role='listbox'>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[0].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[0].commandName +
+						"</button>" +
+						"</li>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[1].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[1].commandName +
+						"</button>" +
+						"</li>" +
+						"<li role='option'>" +
+						"<button aria-label='' class='container toolbar-element' onClick='CKEDITOR.tools.callFunction(" +
+						config.panel.links[2].clickFn +
+						", this" +
+						"); return false;' tablIndex=''>" +
+						config.panel.links[2].commandName +
+						"</button>" +
+						"</li>" +
+						"</ul>"
+				);
+			},
+		},
+	});
+})();
+
+/**
+ * Create a two-dimension array that reflects the actual layout of table cells,
+ * with cell spans, with mappings to the original td elements.
+ *
+ * @param {CKEDITOR.dom.element} table
+ * @member CKEDITOR.tools
+ */
+CKEDITOR.tools.buildTableMap = function (table) {
+	var aRows = table.$.rows;
+
+	// Row and Column counters.
+
+	var r = -1;
+
+	var aMap = [];
+
+	for (var i = 0; i < aRows.length; i++) {
+		r++;
+		if (!aMap[r]) {
+			aMap[r] = [];
+		}
+
+		var c = -1;
+
+		for (var j = 0; j < aRows[i].cells.length; j++) {
+			var oCell = aRows[i].cells[j];
+
+			c++;
+			while (aMap[r][c]) {
+				c++;
+			}
+
+			var iColSpan = isNaN(oCell.colSpan) ? 1 : oCell.colSpan;
+			var iRowSpan = isNaN(oCell.rowSpan) ? 1 : oCell.rowSpan;
+
+			for (var rs = 0; rs < iRowSpan; rs++) {
+				if (!aMap[r + rs]) {
+					aMap[r + rs] = [];
+				}
+
+				for (var cs = 0; cs < iColSpan; cs++) {
+					aMap[r + rs][c + cs] = aRows[i].cells[j];
+				}
+			}
+
+			c += iColSpan - 1;
+		}
+	}
+
+	return aMap;
+};
```

Comment:
Minor SF: Please see our [abbreviation guidelines](https://github.com/liferay/liferay-frontend-projects/blob/dc7d947da927de33e556c7570c007325efa560dc/guidelines/general/naming.md#avoid-abbreviations), almost all variables in this function should be renamed. The code is very poorly readable. For example, `r` should be `row`.

Reply:
I left `i` and `j` as is since is a common convention but I changed:

| Original name | New name           |
|---------------|--------------------|
| aRows         | rows               |
| r             | rowIndex           |
| aMap          | tableMap           |
| c             | columnIndex        |
| oCell         | cellElement        |
| iColSpan      | cellElementColSpan |
| iRowSpan      | rowElementColSpan  |
| rs            | rowSpanCounter     |
| cs            | colSpanCounter     |

Reply:
Comment "ficou lindo" if you agree

Reply:
ficou lindo :100: